rfcs/rfc-ppr-expired-state.md

```diff
@@ -0,0 +1,61 @@
+- Start Date: 2021-11-09
+- Target Major Version: (EPIC or User Story TAG|Link)
+- Reference Issues: (fill in existing related issues, if any)
+- Entity Issue: (leave this empty)
+- Implementation PR: (leave this empty)
+
+# Summary
+
+Build a PPR nightly batch process that runs just after midnight and transitions all database registrations 
+that expired on the previous day to an expired state. 
+- No changes may be made to expired registrations.
+- The scope is PPR API only: regardless of the decision, no change is required to the PPR UI. 
+
+# Motivation
+Implement an efficient solution to setting a registration to an expired state. Most PPR API operations require
+examining and reporting on the registration state; an expired registration is a distinct state.
+Setting the state once and referencing a well-known value for all states is a desirable outcome.
+
+# Detailed design
+
+## Create a PostgreSQL database table
+Record the status of the job in a new log table. This step may be unnecessary if pg_cron is 
+implemented.
+
+## Create a PostgreSQL database function
+This function will transition the financing statement state from active (ACT) to expired (HEX) 
+for all registrations that are active and expired at 23:59:59 Pacific Time on the previous day.
+The function will log the result and return a status.
+Estimated lines of code: < 50.
+
+## Create a scheduled k8s cronjob 
+Run the database function nightly after midnight Pacific Time. Add robustness/reliability including retrying
+a number of times on failure.
+Estimated duration of job execution: < 1 minute.
```

Comment:
My preference would be a script to run the SQL to expire the registrations.
Wrapped as a container, that we can call from k8s cronjob, or as cloud-scheduler/cloudrun without any changes.

Reply:
could we use a trigger function